ts/IdentificationModal.tsx

```diff
@@ -0,0 +1,191 @@
+    return (
+      <React.Fragment>
+        <View spacer={true} extralarge={true} />
+        <TextWithIcon danger={true}>
+          <IconFont name="io-close" color={"white"} />
+          <Text white={true}>{I18n.t("pin_login.pin.confirmInvalid")}</Text>
+        </TextWithIcon>
+      </React.Fragment>
+    );
+  }
+
+  return null;
+};
+
+class IdentificationModal extends React.PureComponent<Props, State> {
+  constructor(props: Props) {
+    super(props);
+
+    this.state = {
+      identificationByPinState: "unstarted"
+    };
+  }
+
+  public render() {
+    const { identification, dispatch } = this.props;
+
+    // If the identification state is started we need to show the modal
+    if (identification.kind === "started") {
+      const {
+        pin,
+        identificationCancelData,
+        identificationSuccessData
+      } = identification;
+
+      const { identificationByPinState } = this.state;
+
+      /**
+       * Create handlers merging default internal actions (to manage the identification state)
+       * with, if available, custom actions passed as props.
+       */
+      const onIdentificationCancelHandler = () => {
+        if (identificationCancelData) {
+          dispatch(identificationCancelData.action);
+        }
+        dispatch(identificationCancel());
+      };
+
+      const onIdentificationSuccessHandler = () => {
+        if (identificationSuccessData) {
+          dispatch(identificationSuccessData.action);
+        }
+        dispatch(identificationSuccess());
+      };
+
+      const onIdentificationFailureHandler = () => {
+        dispatch(identificationFailure());
+      };
+
+      const onPinResetHandler = () => {
+        dispatch(identificationPinReset());
+      };
+
+      return (
+        <Modal
+          visible={true}
+          transparent={true}
+          onRequestClose={onRequestCloseHandler}
+        >
+          <BaseScreenComponent primary={true} contextualHelp={contextualHelp}>
+            <StatusBar barStyle="light-content" />
+            <Content primary={true}>
+              <View spacer={true} extralarge={true} />
+              <Text white={true} alignCenter={true}>
+                {I18n.t("pin_login.pin.pinInfo")}
+              </Text>
+              <Pinpad
+                compareWithCode={pin as string}
+                activeColor={"white"}
+                inactiveColor={"white"}
+                onFulfill={(_: string, __: boolean) =>
+                  this.onPinFullfill(
+                    _,
+                    __,
+                    onIdentificationSuccessHandler,
+                    onIdentificationFailureHandler
+                  )
+                }
+                clearOnInvalid={true}
+              />
+              {renderIdentificationByPinState(identificationByPinState)}
+              <View spacer={true} extralarge={true} />
+              <View>
+                {identificationCancelData !== undefined && (
```

Comment:
you could fallback to a default value for `identificationCancelData` and `onIdentificationCancelHandler` above and avoid the conditional rendering here